modules/platform/nuxeo-platform-usermanager/src/main/java/org/nuxeo/ecm/platform/usermanager/io/NuxeoPrincipalJsonWriter.java

```diff
@@ -111,6 +111,13 @@ public NuxeoPrincipalJsonWriter() {
 
     @Override
     protected void writeEntityBody(NuxeoPrincipal principal, JsonGenerator jg) throws IOException {
+        jg.writeBooleanField("isPartial", false);
+        if (principal.getModel() != null && principal.getModel().getContextData() != null) {
+            Serializable isPartial = principal.getModel().getContextData().get("userHasPartialContent");
+            if (isPartial != null) {
+                jg.writeBooleanField("isPartial", (boolean) isPartial);
+            }
+        }
```

Comment:
Let's not write anything if the user is "normal".
Also the model and the context data are never null.
Finally let's be more robust about the boolean check, check explicitly for true.
```
        Serializable isPartial = principal.getModel().getContextData().get(USER_HAS_PARTIAL_CONTENT);
        if (Boolean.TRUE.equals(isPartial)) {
            jg.writeBooleanField("isPartial", true);
        }
```

Reply:
Also @kevinleturc remarked that this should be moved below, with other `is*` properties.

Reply:
@charlesboidot asked me about the doc being null. 

As we have a nullity check in `writeProperties`, I advised doing the same. 

In the end, I'm in favor of adding the nullity check or remove the one present in `writeProperties`, wdyt?

Reply:
Yes ok for the model we should be a bit careful, as for instance `SystemPrincipal` has a `null` model. Ok to re-add the check:
```
        DocumentModel model = principal.getModel();
        if (model != null) {
            if (Boolean.TRUE.equals(model.getContextData().get(USER_HAS_PARTIAL_CONTENT))) {
                jg.writeBooleanField("isPartial", true);
            }
        }
```